.github/workflows/update.yml

```diff
@@ -25,35 +25,76 @@ jobs:
             branch: 'nixos-24.05'
           - selectedBranch: 'nixos-24.05'
             branch: main
+
     name: Update the flake inputs and generate options
     runs-on: ubuntu-latest
     timeout-minutes: 40
     steps:
       - name: Checkout repository
         uses: actions/checkout@v4
         with:
-          token: ${{ secrets.GH_TOKEN_FOR_UPDATES }}
-          ref: '${{ matrix.branch }}'
+          ref: ${{ matrix.branch }}
 
       - name: Install Nix
         uses: cachix/install-nix-action@v26
         with:
           nix_path: nixpkgs=channel:nixos-unstable
           github_access_token: ${{ secrets.GITHUB_TOKEN }}
 
+      - name: Configure git
+        run: |
+          git config user.name 'github-actions[bot]'
+          git config user.email '41898282+github-actions[bot]@users.noreply.github.com'
+
       - name: Update flake.lock
-        id: update
-        uses: DeterminateSystems/update-flake-lock@v21
+        run: nix flake update --commit-lock-file
+
+      - name: Update autogenerated files
+        run: nix run .#generate-files
+
+      - name: Create Pull Request
+        id: pr
+        uses: peter-evans/create-pull-request@v6
         with:
-          pr-title: "Update flake.lock (${{ matrix.branch }})"
+          # TODO: use an SSH deploy key instead
+          # See https://github.com/peter-evans/create-pull-request/blob/main/docs/concepts-guidelines.md#triggering-further-workflow-runs
+          # and https://github.com/peter-evans/create-pull-request/blob/main/docs/concepts-guidelines.md#push-using-ssh-deploy-keys
           token: ${{ secrets.GH_TOKEN_FOR_UPDATES }}
-          branch: "update_flake_lock_action-${{ matrix.branch }}"
+          git-token: ${{ github.token }}
+          branch: update/${{ matrix.branch }}
+          base: ${{ matrix.branch }}
+          delete-branch: true
+          add-paths: |
+            generated/**
+          commit-message: |
+            generated: Update files
+          title: |
+            [${{ matrix.branch }}] Update flake.lock & generated files
+          # TODO: include commit details in the PR body
+          # Can also use body-path if it's easier to generate a file
+          # body: |
```

Comment:
We can also leave it for another PR.

Reply:
TBH, I'm not sure it's worth the effort. Should I just remove the comments?

Reply:
I don't know. The action we currently use does it automatically right ?
But I agree, not a huge deal.

Reply:
It shouldn't be too hard to print the flake.lock commit message and then also the files changed by the auto-gen script, I'll try and find motivation to come up with something.

Reply:
We can get the commit body without title using:
```shell
git show --no-patch --format=%b
```
(or `%B` for full body with title)

As for the auto-generated files, we can get the unstaged changes with:
```shell
git diff --name-status
# or
git diff --name-only
```

Although this will only list _tracked_ files with unstaged changes.

To get any file that `git add -A` would add, we could use the `--dry-run` option, e.g:
```shell
$ git add -A --dry-run ./generated
add 'generated/efmls-configs.nix'
add 'generated/foo'
```

Reply:
Ok ! That looks promising !
No need to go crazy about it either. I guess it's a cool bonus.